### DISCUSSION TITLE
GraphQL should allow to upload files

### DISCUSSION THREAD

**Discussion by moekify** [directus/directus#5313] [Feature Requests]
Hi!

I was looking into using the GraphQL endpoint to manage form responses. 
More specifically, the option to upload files via GQL.
I'm aware that the same can be done with the REST API, but I would be curious to hear some opinion on enabling this via GQL.

General Idea of GQL file uploads: https://github.com/jaydenseric/graphql-multipart-request-spec

I'm not a professional in GQL matters and wouldn't necessary be confident in implementing this myself. Even though I'd be happy to give it a shot if no one else cares, but it still being an interesting feature to have.

We probably would add something here: https://github.com/directus/directus/blob/main/api/src/services/graphql.ts#L1930

Relevant: https://graphql-compose.github.io/docs/guide/file-uploads.html

**Comment by rijkvanzanten** [comment]
From the source you linked yourself:

> If you decide how to upload files via some REST endpoint or GraphQL. So I recommend to upload via some REST API and then provide a path of the uploaded file to your mutation request. GraphQL designed to provide typed data according to client request shape. With files (binary data) it works too, but better to do it via well-recommended REST calls. In such case, you separate highly costed upload logic from data manipulation logic. In the future, this will help you diagnose problems with the load more easily.

😄

GraphQL doesn't seem to have any native way to go about this, and that spec seems to be non-official, meaning that I'm not sure how support for this would be in other graphql tooling 🤔

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
